go_sequencer/common/score.go

```diff
@@ -1,10 +1,48 @@
 package common
 
+import (
+	"encoding/binary"
+	"math/big"
+)
+
 // Score is a struct that gives an information about score
 // of each accounts.
 type Score struct {
 	Idx      AccountIdx `meddler:"idx"`
 	BatchNum BatchNum   `meddler:"batch_num"`
-	Nonce    Nonce      `meddler:"-"` // max of 40 bits used
-	Score    bool       `meddler:"score"`
+	Value    uint64     `meddler:"score"`
+}
+
+const (
+	// maxScore is the maximum value that Score can have (30 bits:
+	// maxScoreValue=2**30-1)
+	maxScoreValue = 0xffffffff
+)
+
+// Bytes returns a byte array representing the score
+func (s *Score) Bytes() ([4]byte, error) {
+	if s.Value > maxScoreValue {
+		return [4]byte{}, Wrap(ErrScoreOverflow)
+	}
+	var scoreBytes [8]byte
+	binary.BigEndian.PutUint64(scoreBytes[:], uint64(s.Value))
+	var b [4]byte
+	copy(b[:], scoreBytes[4:])
+	return b, nil
+}
+
+// ScoreFromBytes returns score from a byte array
+func ScoreFromBytes(b [4]byte) (*Score, error) {
+	var scoreBytes [8]byte
```

Comment:
Will it be of [4]byte instead of [8] ?

Reply:
It is not possible, since uint64 has input as [8]bytes, if I changed to [4]bytes, there is nil pointer error.

Reply:
Instead of uint64, use uint32 for score.

Reply:
Makes, sense should we update score - Value to uint32 instead since it'll be of 30 bit